### PR TITLE
#5 - Events

### PR DESCRIPTION
**Pull request by rblalock**
Addresses #5 , #2   - This PR adds necessary events and callbacks needed to manage the blocks in the notebook

### CODE REVIEW

**Comment by rblalock** [comment]
I guess I need to add some things to .gitignore.....

File: src/BlockEditorContext.tsx [hunk 1]
```diff
@@ -14,21 +16,48 @@ export interface BlockEditorProviderProps {
...
+	// Handles reordering the blocks (drag-n-dropped)
 	const reorder = (startIndex: number, endIndex: number) => {
 		const orderedBlocks = util.reorder(blocks, startIndex, endIndex);
 		setBlocks(orderedBlocks);
 
 		props.onReorder && props.onReorder(orderedBlocks, startIndex, endIndex);
 	};
 
+	const findBlock = (id: string) => {
+		const block = blocks.filter((b) => `${b.id}` === id)[0];
```
Comment (chrisbowley) [inline src/BlockEditorContext.tsx:33]: What not use `.find()`?

Comment (chrisbowley) [reply src/BlockEditorContext.tsx:33]: `${b.id}` - is this to guard against an empty `id`?

File: src/BlockEditorContext.tsx [hunk 2]
```diff
@@ -14,21 +16,48 @@ export interface BlockEditorProviderProps {
...
+
+		setBlocks(
+			update(blocks, {
+				$splice: [
+					[index, 1, block],
+				],
+			}),
+		);
+
+		if (props.onEditBlock) props.onEditBlock(block, index);
```
Comment (chrisbowley) [inline src/BlockEditorContext.tsx:53]: Inline if statement? 😱

Comment (rblalock) [reply src/BlockEditorContext.tsx:53]: Is that against the Chris Bowley linter?

File: src/BlockEditorContext.tsx (hunk 1 above)
Comment (rblalock) [reply src/BlockEditorContext.tsx:33]: It's a port from the UIC library and I don't recall the reason but it made UIC happy

**Review by chrisbowley — APPROVED** [review]

**Pull request merged**